src/Schema/Events/Experiment.ts

```diff
@@ -0,0 +1,46 @@
+import { OwnerType } from "../Values/OwnerType"
+import { ActionType } from "."
+
+/**
+ * Schemas describing experiment events
+ * @packageDocumentation
+ */
+
+/**
+ * A user received an experiment on Artsy
+ *
+ * This schema describes events sent to Segment from [[Experiment]]
+ *
+ *  @example
+ *  ```
+ *  {
+ *    action: "experiment",
+ *    event: "get_variant",
+ *    service: "unleash",
+ *    experiment_name: "some_experiment_we_are_running",
+ *    variant_name: "some_variant_for_this_user"
+ *    payload: "some_payload",
+ *    context_owner_type: "artwork",
+ *    context_owner_id: "55ed8ca57261693d930000b8",
+ *  }
+ * ```
+ */
+export type Experiment = {
+  action: ActionType.experiment
+  service: "unleash" | string
+  experiment_name: string
+  enabled: string
+
+  context_owner_type: OwnerType
+  context_owner_id?: string
+  context_owner_slug?: string
+} & (
+  | {
+      event: "get_flag"
```

Comment:
How does this part work? It's not a property is it?

Reply:
you mean the `&` and `|` part?

everything before `&` is what is needed, and then also everything after is needed. after `&` we have a paren with two options, marked with `|`, so one or the other is added.

so with that, we end up with a type that would allow us something like that:
```
{
    action: "experiment",
    event: "get_variant",
    service: "unleash",
    experiment_name: "some_experiment_we_are_running",
    variant_name: "some_variant_for_this_user"
    payload: "some_payload",
    context_owner_type: "artwork",
    context_owner_id: "55ed8ca57261693d930000b8",
}
```

it will allow:
```
{
    action: "experiment",
    event: "get_flag",
    service: "unleash",
    experiment_name: "some_experiment_we_are_running",
    context_owner_type: "artwork",
    context_owner_id: "55ed8ca57261693d930000b8",
}
```

but it will *not* allow this:
```
{
    action: "experiment",
    event: "get_variant",
    service: "unleash",
    experiment_name: "some_experiment_we_are_running",
// notice that `variant_name` is missing here!
    context_owner_type: "artwork",
    context_owner_id: "55ed8ca57261693d930000b8",
}
```

does that make sense?